src/realm/object-store/c_api/sync.cpp

```diff
@@ -410,6 +411,69 @@ RLM_API void realm_sync_config_set_resync_mode(realm_sync_config_t* config,
     config->client_resync_mode = ClientResyncMode(mode);
 }
 
+RLM_API realm_object_id_t realm_flx_sync_subscription_id(const realm_flx_sync_subscription_t* subscription) noexcept
+{
+    REALM_ASSERT(subscription != nullptr);
+    return wrap_err([&]() {
+        return to_capi(subscription->id());
+    });
+}
+
+RLM_API bool realm_flx_sync_subscription_has_name(const realm_flx_sync_subscription_t* subscription) noexcept
+{
+    REALM_ASSERT(subscription != nullptr);
+    return wrap_err([&]() {
+        return subscription->has_name();
+    });
+}
+
+RLM_API const char* realm_flx_sync_subscription_name(const realm_flx_sync_subscription_t* subscription) noexcept
+{
+    REALM_ASSERT(subscription != nullptr);
+    return wrap_err([&]() {
+        if (subscription->has_name()) {
+            return subscription->name().data();
+        }
+        return (const char*)nullptr;
```

Comment:
It will, but .data() will also return nullptr, so I can simplify. It is an error to access name(), if has_name() returns false.

Reply:
All of these methods can fail if the SubscriptionSet is closed (I assume?) So wouldn't we need to turn the return value into a boolean and having `out_*` params for all of them?

Reply:
They use `wrap_err` and return null on any error. You can get the error by calling `realm_get_last_error`.

Reply:
The discussion is more on what to return, if name is not given. As is we return null, which should be considered an error on the SDK. My argument is that, that is okay, since the user should call `has_name` first to determine if a call to `name` should even be considered.

Reply:
I don't think we need two methods to achieve a simple property lookup. It's not a hot path, but we shouldn't be introducing inefficiencies like that.

Reply:
@cmelchior all the methods in question throw no errors at all: https://github.com/realm/realm-core/blob/ab94383d032d1fb1e0d9203cb31fa32334f996aa/src/realm/sync/subscriptions.cpp#L80-L116

I don't think we need error handling for these at all. They should not be using `wrap_err` and they should be marked with `RLM_API_NOEXCEPT` to indicate they are not expected to throw any errors a developer can handle.

Reply:
Have updated PR, as suggested by @fealebenpae (ie. remove superfluous `wrap_err` calls).